### ISSUE TITLE
Create symbolic link inside container before starting it

### ISSUE THREAD

**Issue by johnlane** [terraform-lxd/terraform-provider-lxd#256]
I'm looking for a way to create a symbolic link inside a contaier before it is started, similar to how files may be copied in.

There isn't an equivalent `lxc` command for this but [it can be done](https://discuss.linuxcontainers.org/t/create-symlink-in-stopped-container/13498) using `curl`:

```
curl --unix-socket /var/lib/lxd/unix.socket -s lxd/1.0/containers/mycontainer/files?path=/etc/runlevels/default/firstboot -H "X-LXD-type: symlink" -H "Content-Type: text/plain" -d "/etc/init.d/firstboot"
```

I don't know if it's possible to execute such a `curl` command using Terraform? It would need to execute on the lxd host - which may not be (and is not, in my case) where Terraform is run. I thought about `local_exec` (runs on TF host) and `remote_exec` (runs inside running container over ssh) - neither of which are suitable for this.

My specific use-case is to set up a firstboot script in an Alpine Linux container which needs to be done before the container starts. I'm using the upstream Alpine image and would prefer to avoid maintaining my own image.

**Comment by adamcstephens** [comment]
It looks like [CreateInstanceFile](https://pkg.go.dev/github.com/lxc/lxd/client#ProtocolLXD.CreateInstanceFile) can take a file arg type of "symlink" even if [undocumented](https://pkg.go.dev/github.com/lxc/lxd/client#InstanceFileArgs). Unfortunately, the `lxd_container_file` resource isn't currently flexible enough to accommodate such a call and would need to be refactored.

Closing this as won't fix for now. There are alternatives, including cloud-init and mastering your own images. I'd be happy to discuss if you'd like to create a PR, or feel free to reply or reopen.
